cmd/agentsecret_cmd.go

```diff
@@ -0,0 +1,29 @@
+package cmd
+
+import (
+	"github.com/ingrammicro/cio/api/agentsecret"
+	"github.com/ingrammicro/cio/utils"
+	"github.com/ingrammicro/cio/utils/format"
+	"github.com/urfave/cli"
+)
+
+// WireUpSecret prepares common resources to send request to API
+func WireUpSecret(c *cli.Context) (ss *agentsecret.SecretService, config *utils.Config, f format.Formatter) {
+
+	f = format.GetFormatter()
+
+	config, err := utils.GetConcertoConfig()
+	if err != nil {
+		f.PrintFatal("Couldn't wire up config", err)
+	}
+	hcs, err := utils.NewHTTPConcertoService(config)
+	if err != nil {
+		f.PrintFatal("Couldn't wire up concerto service", err)
+	}
+	ss, err = agentsecret.NewSecretService(hcs, config.APIEndpoint)
```

Comment:
It's valid for me, just a style point of view. This will be refactored and unified in the future.
Since "config" is not visible from NewHTTPConcertoService; in some case we are returning config and then/later passing it to end method which manage the petition.

** It's a minor comment, the idea is to refactor and unify all of this in the future( with branch: feature/46-migrate-to-cobra)

Reply:
I know it's done differently with attachments for example, but I think this way provides a better separation of concerns: - If  these services can know the base API endpoint for requests other than file requests then it follows that they should know the base endpoint for a file request when it is the API endpoint and it does not depend on input parameters.
- Similarly, if some piece of code using one these services does not need to know the API endpoint for non-file requests, it should not need to know it for file-requests that use it as API endpoint...

But I think the problem is that `utils.ConcertoService.GetFile` requires as parameter a complete URL instead of accepting a path. If that were implemented, this line could just be:
 ```go
ss, err = agentsecret.NewSecretService(hcs)
 ```
 which makes much more sense.